daml-lf/engine/src/main/scala/com/digitalasset/daml/lf/engine/Error.scala

```diff
@@ -45,6 +45,11 @@ final case class ContractNotFound(ci: ContractId) extends Error {
   override def detailMsg: String = msg
 }
 
+final case class ContractKeyNotFound(key: GlobalKey) extends Error {
```

Comment:
@daravep fyi, this also covers a case that your string matching misses I believe (see the two different tests we have).

Reply:
Thanks a lot! My string matching is a "first step", reverse engineering the possible errors using the ledger API conformance tests. @remyhaemmerle-da is rewriting the lf-errors into case classes so we don't have to match the strings again. Please feel free to correct and fix the errors and change the strings.